### PR TITLE
Add colour customisation to layout editor

### PR DESCRIPTION
**Pull request by u4vh3**
Allows you to change colour for:
- Argon wedge piece
- Box element
- Default and argon song progress
- Font adjustable components

https://github.com/user-attachments/assets/e8bdfb27-0bd5-403c-bd98-6a31f0f45b8f

I also implemented this to `FontAdjustableSkinComponent` because it makes sense but I'm not too sure.

I don't think there's design for button for adjusting colour, so I came up with my own.

### CODE REVIEW

File: osu.Game.Tests/Visual/UserInterface/TestSceneSettingsColour.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+#nullable disable
```
Comment (bdach) [inline osu.Game.Tests/Visual/UserInterface/TestSceneSettingsColour.cs:4]: `#nullable disable` is not allowed in any new code

File: osu.Game/Screens/Play/HUD/ArgonSongProgress.cs [hunk 2]
```diff
@@ -29,6 +30,9 @@ public partial class ArgonSongProgress : SongProgress
         [SettingSource(typeof(SongProgressStrings), nameof(SongProgressStrings.ShowTime), nameof(SongProgressStrings.ShowTimeDescription))]
         public Bindable<bool> ShowTime { get; } = new BindableBool(true);
 
+        [SettingSource(typeof(SkinnableComponentStrings), nameof(SkinnableComponentStrings.Colour), nameof(SkinnableComponentStrings.ColourDescription))]
+        public new BindableColour4 Colour { get; } = new BindableColour4(Colour4.White);
```
Comment (bdach) [inline osu.Game/Screens/Play/HUD/ArgonSongProgress.cs:34]: Not a fan of the shadowing here (the usage of `new`) or the usage of `base` underneath. I'd rename all of these to `AccentColour` which gets rid of both.

File: osu.Game/Skinning/FontAdjustableSkinComponent.cs [hunk 3]
```diff
@@ -20,11 +21,16 @@ public abstract partial class FontAdjustableSkinComponent : Container, ISerialis
         [SettingSource(typeof(SkinnableComponentStrings), nameof(SkinnableComponentStrings.Font), nameof(SkinnableComponentStrings.FontDescription))]
         public Bindable<Typeface> Font { get; } = new Bindable<Typeface>(Typeface.Torus);
 
+        [SettingSource(typeof(SkinnableComponentStrings), nameof(SkinnableComponentStrings.FontColour), nameof(SkinnableComponentStrings.FontColourDescription))]
+        public BindableColour4 FontColour { get; } = new BindableColour4(Colour4.White);
```
Comment (bdach) [inline osu.Game/Skinning/FontAdjustableSkinComponent.cs:25]: Please rename this (and all associated members) to `TextColour`. Fonts don't have colours.

**Review by bdach — CHANGES_REQUESTED** [review]

**Review by bdach — DISMISSED** [review]
sure I guess?

File: osu.Game/Screens/Play/HUD/ArgonSongProgress.cs [hunk 4]
```diff
@@ -114,6 +118,7 @@ protected override void Update()
             base.Update();
             content.Height = bar.Height + bar_height + info.Height;
             graphContainer.Height = bar.Height;
+            Colour = AccentColour.Value;
```
Comment (peppy) [inline osu.Game/Screens/Play/HUD/ArgonSongProgress.cs:121]: I question why this needs to be in update. Let's use the bindable as a bindable instead yeah?

File: osu.Game/Screens/Play/HUD/DefaultSongProgress.cs [hunk 5]
```diff
@@ -114,6 +118,8 @@ protected override void Update()
 
             if (!Precision.AlmostEquals(Height, newHeight, 5f))
                 content.Height = newHeight;
+
+            Colour = AccentColour.Value;
```
Comment (peppy) [inline osu.Game/Screens/Play/HUD/DefaultSongProgress.cs:122]: Same here.

**Review by peppy — CHANGES_REQUESTED** [review]
Seems pretty okay otherwise.

File: osu.Game/Screens/Play/HUD/ArgonSongProgress.cs (hunk 4 above)
Comment (bdach) [reply osu.Game/Screens/Play/HUD/ArgonSongProgress.cs:121]: jesus. not sure how i'd missed that.

Comment (bdach) [reply osu.Game/Screens/Play/HUD/ArgonSongProgress.cs]: removed in https://github.com/ppy/osu/pull/29936/commits/9f4e48dde78eb1b76c6131999aa21aeb4dc42843. also found one more when i was looking at that again.

**Review by peppy — APPROVED** [review]

**Pull request merged**